tests/e2e/util/kubectl/kubectl.go

```diff
@@ -21,97 +21,91 @@ import (
 	"time"
 
 	"github.com/istio-ecosystem/sail-operator/tests/e2e/util/shell"
+	"github.com/onsi/ginkgo/v2"
 )
 
+type KubectlBuilder struct {
+	binary     string
+	namespace  string
+	kubeconfig string
+}
+
 const DefaultBinary = "kubectl"
 
-// optionalKubeconfig add the flag --kubeconfig if the kubeconfig is set
-func optionalKubeconfig(kubeconfig []string) string {
-	if len(kubeconfig) > 0 && kubeconfig[0] != "" {
-		return fmt.Sprintf("--kubeconfig %s", kubeconfig[0])
-	}
-	return ""
+func newKubectlBuilder() *KubectlBuilder {
+	return &KubectlBuilder{}
 }
 
-// kubectl return the kubectl command
-// If the environment variable COMMAND is set, it will return the value of COMMAND
-// Otherwise, it will return the default value "kubectl" as default
-// Arguments:
-// - format: format of the command without kubeclt or oc
-// - args: arguments of the command
-func kubectl(format string, args ...interface{}) string {
+func (k *KubectlBuilder) setBinary() {
 	binary := DefaultBinary
 	if cmd := os.Getenv("COMMAND"); cmd != "" {
 		binary = cmd
 	}
 
-	return binary + " " + fmt.Sprintf(format, args...)
+	k.binary = binary
 }
 
-// CreateFromString creates a resource from the given yaml string
-func CreateFromString(yamlString string, kubeconfig ...string) error {
-	cmd := kubectl("create %s -f -", optionalKubeconfig(kubeconfig))
-	_, err := shell.ExecuteCommandWithInput(cmd, yamlString)
-	if err != nil {
-		return fmt.Errorf("error creating resource from yaml: %w", err)
-	}
-	return nil
+func (k *KubectlBuilder) setNamespace() {
+	k.namespace = ""
 }
 
-// ApplyString applies the given yaml string to the cluster
-func ApplyString(ns, yamlString string, kubeconfig ...string) error {
-	nsflag := nsflag(ns)
-	// If the namespace is empty, we need to remove the flag because it will fail
-	// TODO: improve the nsflag function to handle this case
-	if ns == "" {
-		nsflag = ""
+func (k *KubectlBuilder) setKubeconfig() {
```

Comment:
do we really need this func? kubeconfig fields will be empty after initialization, right?